### PR TITLE
feat: add test coverage with Vitest, CI workflow, and switch to pnpm

### PR DESCRIPTION
**Pull request by schalkneethling**
## Summary

- Set up **Vitest** with jsdom + **React Testing Library** as the test framework
- Extract helper functions (`daysSince`, `fmtDate`, `fmtDuration`, `timeAgo`) from `App.jsx` into `src/lib/helpers.js` for testability and reuse
- Add **33 tests** across 4 test files:
  - `helpers.test.js` (14) — edge cases, formatting, time calculations
  - `useAuth.test.js` (7) — session loading, OAuth flow, subscription cleanup
  - `useSettings.test.js` (6) — token detection, upsert, no-op on null/empty
  - `LoginScreen.test.jsx` (5) — rendering, loading/disabled state, click handler
- Add shared **Supabase client mock** (`src/lib/__mocks__/supabase.js`)
- Add **GitHub Actions CI** workflow — runs tests and build on push/PR to main
- Switch from **npm to pnpm** (lockfile, CI workflow using `pnpm/action-setup@v4`)

## Test plan

- [x] All 33 tests pass locally (`pnpm test`)
- [x] Build succeeds (`pnpm build`)
- [ ] CI workflow runs successfully on this PR

### CODE REVIEW

File: src/lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ? new Date(d).toLocaleDateString("en-ZA", {
+        day: "numeric",
+        month: "short",
+        year: "numeric",
+      })
+    : "";
+
+export const fmtDateTime = (d) =>
+  d
+    ? new Date(d).toLocaleDateString("en-ZA", {
```
Comment (schalkneethling) [inline src/lib/helpers.js]: We should use the user's locale if we can retrieve it and only use en-ZA as a fallback.

File: src/lib/helpers.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export const timeAgo = (d) => {
+  if (!d) return "";
+  const s = Math.floor((Date.now() - new Date(d).getTime()) / 1000);
+  if (s < 60) return "just now";
+  if (s < 3600) return `${Math.floor(s / 60)}m ago`;
+  if (s < 86400) return `${Math.floor(s / 3600)}h ago`;
+  const days = Math.floor(s / 86400);
+  return days === 1 ? "yesterday" : `${days}d ago`;
+};
```
Comment (schalkneethling) [inline src/lib/helpers.js:44]: Could we simplify or perhaps even not need any of this if we use the new Temporal with the official Temporal polyfill for browsers that do not yet support it?
